libs/wire-api/src/Wire/API/User/Identity/AuthId.hs

```diff
@@ -114,21 +112,25 @@ data EmailSource
 -- data type and everyhing that's using it.
 newtype LegacyAuthId = LegacyAuthId {fromLegacyAuthId :: TeamId -> AuthId}
 
+authIdEmailWithSource :: AuthId -> Maybe EmailWithSource
+authIdEmailWithSource = \case
+  AuthSAML _ -> Nothing
+  AuthSCIM (ScimDetails _ ews) -> Just ews
+  AuthBoth _ _ mbEws -> mbEws
+
 -- | 'SAML.UserRef' and 'externalId' are isomorphic.  (Both DB indices in spar are kept in
 -- sync: if both exist and identify the same user, they do so via those `spar.user` and
 -- `spar.scim_external`.)
 --
--- The 'Nothing' case can occur if the 'NameID' has unsupported qualifiers.
---
 -- (The 'TeamId' could be inferred from the saml issuer, but that would be a database lookup,
 -- which would make the function even less total.)
-urefToExternalId :: TeamId -> SAML.UserRef -> Maybe ExternalId
-urefToExternalId tid uref = ExternalId tid <$> SAML.shortShowNameID (uref ^. SAML.uidSubject)
-
--- | See 'userRefToExternalId'.  'NameID' format is either email (if parseable) or unspecified
--- (if not).
-externalIdToUref :: SAML.Issuer -> ExternalId -> SAML.UserRef
-externalIdToUref = undefined
+authIdExternalId :: TeamId -> AuthId -> Maybe ExternalId
```

Comment:
Unused. Remove
- [ ] done

Reply:
It is used again, but it shouldn't be, I think.

It's turning an `AuthSAML` into an `ExternalId`, which (I think) is only needed for cases where a SAML-only user is getting in contact with SCIM for the first time.  After that, we should always end up in the `AuthBoth` case.  So when connecting a user to SCIM, instead of this function, we should do something that makes it more explicit that we're creating a fresh `ExternalId`.

Also, the `AuthBoth` case is not using the `tid` it already carries,  but one it is being passed as an argument.  This is at best a bad smell.

Reply:
I inlined the funciton and removed it from wire-api